src/audio/module_adapter/module/volume/volume.c

```diff
@@ -941,13 +946,20 @@ static int volume_set_config(struct processing_module *mod, uint32_t config_id,
 	cdata = *(const struct ipc4_peak_volume_config *)fragment;
 	cdata.target_volume = convert_volume_ipc4_to_ipc3(dev, cdata.target_volume);
 
+	if ((cdata.channel_id != IPC4_ALL_CHANNELS_MASK) &&
+	    (cdata.channel_id >= SOF_IPC_MAX_CHANNELS))
+		return -EINVAL;
+
 	init_ramp(cd, cdata.curve_duration, cdata.target_volume);
 	cd->ramp_finished = true;
 
+	channels_count = mod->priv.cfg.base_cfg.audio_fmt.channels_count;
+	assert(channels_count < SOF_IPC_MAX_CHANNELS);
```

Comment:
Why assert here? Another check looks different
`if (channels_count >= SOF_IPC_MAX_CHANNELS)`

Reply:
I think that according to previous check (channels_count <= SOF_IPC_MAX_CHANNELS) should be asserted

Reply:
I mean why we have _assert_ here instead of the same _if_,
FW should return error if set config failed instead of crash.

Reply:
ack, this is non fatal (just a bad topology) so FW should not crash but report up the stack.